Autoconf and automake upgrades caused trouble for packages buildbot

### Describe the bug

Looks like the recent upgrade of autoconf by #17825 and automake by #17827 (and maybe some other toolchain tools at the same time) have caused major trouble for the packages buildbot. The current faillogs list is 400+ packages, including procd, firewall, etc. basic OpenWrt packages.

The errors seem to relate the autoconf/automake components, or finding libtool etc.

Examples from buildbot faillogs today:

https://downloads.openwrt.org/snapshots/faillogs/aarch64_cortex-a76/
https://downloads.openwrt.org/snapshots/faillogs/arm_cortex-a7/base/
https://downloads.openwrt.org/snapshots/faillogs/mips64_mips64r2/

wolfssl:
```
autoreconf: export WARNINGS=
autoreconf: Entering directory '.'
autoreconf: configure.ac: not using Gettext
autoreconf: running: /builder/shared-workdir/build/sdk/staging_dir/host/bin/aclocal -I /builder/shared-workdir/build/sdk/staging_dir/target-arm_cortex-a7_musl_eabi/host/share/aclocal -I /builder/shared-workdir/build/sdk/staging_dir/hostpkg/share/aclocal -I /builder/shared-workdir/build/sdk/staging_dir/target-arm_cortex-a7_musl_eabi/usr/share/aclocal -I m4 -I . --force -I m4
/builder/shared-workdir/build/sdk/staging_dir/host/bin/aclocal: 2: /builder/shared-workdir/build/sdk/staging_dir/host/bin/aclocal.real: not found
autoreconf: error: /builder/shared-workdir/build/sdk/staging_dir/host/bin/aclocal failed with exit status: 127
/builder/shared-workdir/build/sdk/staging_dir/host/bin/sed: can't read /builder/shared-workdir/build/sdk/build_dir/target-arm_cortex-a7_musl_eabi/wolfssl-regular/wolfssl-5.7.6-stable/libtool: No such file or directory
```

libmnl
```
CDPATH="${ZSH_VERSION+.}:" && cd . && /bin/bash '/builder/shared-workdir/build/sdk/build_dir/target-arm_cortex-a7_musl_eabi/libmnl-1.0.5/build-aux/missing' aclocal-1.16 -I m4
/builder/shared-workdir/build/sdk/staging_dir/host/bin/aclocal-1.16: 2: /builder/shared-workdir/build/sdk/staging_dir/host/bin/aclocal.real: not found
WARNING: 'aclocal-1.16' is missing on your system.
         You should only need it if you modified 'acinclude.m4' or
         'configure.ac' or m4 files included by 'configure.ac'.
         The 'aclocal' program is part of the GNU Automake package:
         <https://www.gnu.org/software/automake>
         It also requires GNU Autoconf, GNU m4 and Perl in order to run:
         <https://www.gnu.org/software/autoconf>
         <https://www.gnu.org/software/m4/>
         <https://www.perl.org/>
make[4]: *** [Makefile:433: aclocal.m4] Error 127
```

asterisk
```
Can't locate Autom4te/ChannelDefs.pm in @INC (you may need to install the Autom4te::ChannelDefs module) (@INC contains: /builder/shared-workdir/build/staging_dir/host/share/autoconf /etc/perl /usr/local/lib/x86_64-linux-gnu/perl/5.32.1 /usr/local/share/perl/5.32.1 /usr/lib/x86_64-linux-gnu/perl5/5.32 /usr/share/perl5 /usr/lib/x86_64-linux-gnu/perl-base /usr/lib/x86_64-linux-gnu/perl/5.32 /usr/share/perl/5.32 /usr/local/lib/site_perl) at /builder/shared-workdir/build/sdk/staging_dir/host/bin/autoconf line 42.
BEGIN failed--compilation aborted at /builder/shared-workdir/build/sdk/staging_dir/host/bin/autoconf line 42.
Please install autoconf and run bootstrap.sh again!
```


cc @Ansuel 


### OpenWrt version

main

### OpenWrt release

main

### OpenWrt target/subtarget

all

### Device

toolchain

### Image kind

Official downloaded image

### Steps to reproduce

_No response_

### Actual behaviour

_No response_

### Expected behaviour

_No response_

### Additional info

_No response_

### Diffconfig

```text

```

### Terms

- [x] I am reporting an issue for OpenWrt, not an unsupported fork.

--- a/tools/autoconf/patches/000-relocatable.patch
+++ b/tools/autoconf/patches/000-relocatable.patch
@@ -1,3 +1,34 @@
+--- a/bin/autoconf.in
++++ b/bin/autoconf.in
+@@ -29,7 +29,10 @@ use warnings FATAL => 'all';
+ 
+ BEGIN
+ {
+-  my $pkgdatadir = $ENV{'autom4te_perllibdir'} || '@pkgdatadir@';
++  my $pkgdatadir = $ENV{'autom4te_perllibdir'} ||
++		($ENV{'STAGING_DIR_HOST'} ?
++			$ENV{'STAGING_DIR_HOST'} . '/share/autoconf' :
++			'@pkgdatadir@');
+   unshift @INC, $pkgdatadir;
+ 
+   # Override SHELL.  On DJGPP SHELL may not be set to a shell
+@@ -44,8 +47,14 @@ use Autom4te::Channels qw(msg);
+ use Autom4te::General;
+ 
+ # Lib files.
+-my $autom4te = $ENV{'AUTOM4TE'} || '@bindir@/@autom4te-name@';
+-my $trailer_m4 = $ENV{'trailer_m4'} || '@pkgdatadir@/autoconf/trailer.m4';
++my $autom4te = $ENV{'AUTOM4TE'} ||
++	($ENV{'STAGING_DIR_HOST'} ?
++		$ENV{'STAGING_DIR_HOST'} . '/bin/@autom4te-name@' :
++		'@bindir@/@autom4te-name@');
++my $trailer_m4 = $ENV{'trailer_m4'} ||
++	($ENV{'STAGING_DIR_HOST'} ?
++		$ENV{'STAGING_DIR_HOST'} . '/share/autoconf/autoconf/trailer.m4' :
++		'@pkgdatadir@/autoconf/trailer.m4');
+ 
+ # $HELP
+ # -----
 --- a/bin/autoheader.in
 +++ b/bin/autoheader.in
 @@ -30,9 +30,12 @@ use 5.006;

